.ddev/ci/matrix.py

```diff
@@ -0,0 +1,190 @@
+        if not ((directory / 'hatch.toml').is_file() and (directory / 'tests').is_dir()):
+            continue
+
+        for relative_path in files:
+            possible_file = directory / relative_path
+            if possible_file.name in NON_TESTABLE_FILES:
+                continue
+            elif any(fnmatch(str(possible_file), str(directory / pattern)) for pattern in TESTABLE_FILE_PATTERNS):
+                testable_directories.append(directory_name)
+                break
+
+    testable_directories.sort(key=lambda d: (DISPLAY_ORDER_OVERRIDE.get(d, len(DISPLAY_ORDER_OVERRIDE)), d))
+
+    overrides = {}
+    if (config_file := root / '.ddev' / 'config.toml').is_file():
+        overrides.update(tomllib.loads(config_file.read_text(encoding='utf-8')).get('overrides', {}))
+
+    display_overrides = overrides.get('display-name', {})
+    ci_overrides = overrides.get('ci', {})
+
+    matrices = []
+    for directory_name in testable_directories:
+        manifest = read_manifest(root, directory_name)
+
+        matrix_overrides = ci_overrides.get(directory_name, {})
+        platforms = matrix_overrides.get('platforms', [])
+        if not platforms:
+            if manifest:
+                platforms = []
+                for classifier_tag in manifest['tile']['classifier_tags']:
+                    key, _, value = classifier_tag.partition('::')
+                    if key == 'Supported OS':
+                        platforms.append(value.lower())
+
+                # Run only on Linux if not exclusive to Windows
+                if platforms != ['windows']:
+                    platforms = ['linux']
+            else:
+                platforms = ['linux']
+
+        for platform in platforms:
+            if platform not in PLATFORM_NAMES:
+                raise ValueError(f'Unsupported platform for `{directory_name}`: {platform}')
+
+            config = {'os': PLATFORM_TO_OS[platform], 'target': directory_name}
+
+            if directory_name in display_overrides:
+                config['name'] = display_overrides[directory_name]
+            elif manifest:
+                config['name'] = manifest['assets']['integration']['source_type_name']
+            else:
+                config['name'] = directory_name
+
+            if len(platforms) > 1:
+                config['name'] += f' on {PLATFORM_NAMES[platform]}'
+
+            if matrix_overrides.get('test-py2', False):
+                config['py2'] = True
+
+            matrices.append(config)
+
+    total_matrices = len(matrices)
+    matrices[:] = matrices[:JOB_LIMIT]
+
+    output = json.dumps(matrices, indent=2) if args.pretty else json.dumps(matrices, separators=(',', ':'))
+    print(output)
+
+    print(f'Output size: {len(output)} / {OUTPUT_LIMIT}', file=sys.stderr)
+    print(f'Number of jobs: {total_matrices}', file=sys.stderr)
+    if total_matrices > JOB_LIMIT:
+        print(f'Jobs ignored to satisfy limit: {total_matrices - JOB_LIMIT}', file=sys.stderr)
```

Comment:
What is the expected action item in this situation? Do I understand correctly that the ignored jobs are never run?

Reply:
Yes but right now we are at 2/3 of the limit so we won't hit it for a long time and also it would be rare to modify that many files at once. Furthermore if the base or dev packages are modified this workflow does not run and the other one that tests everything with a giant file of enumerated jobs runs